### PR TITLE
Make OtlpExporter class internal

### PR DESCRIPTION
**Pull request by alanwest**
The constructor on all of our exporter classes is internal. I think making the class internal is the right thing to do.  If so, I can expand this PR to make this change to zipkin, jaeger, etc.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol
     /// Exporter consuming <see cref="Activity"/> and exporting the data using
     /// the OpenTelemetry protocol (OTLP).
     /// </summary>
-    public class OtlpExporter : BaseExporter<Activity>
+    internal class OtlpExporter : BaseExporter<Activity>
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs:35]: the ctor() was already internal, so we wont break anyone right?

Comment (alanwest) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs:35]: By the class being public, the type is visible, so a user can type `var exporter = new OtlpExporter()` ... but with no way to construct it, is there any value in exposing it?

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs:35]: Agree.
I was thinking -- if a user want to write own Batching/SimpleExportprocessor, then they wont be able to do that, as they cannot construct the exporter. But its okay, as spec talks about only 2 ExportingProcessors. and https://github.com/open-telemetry/opentelemetry-dotnet/pull/1504 will solve customization issues as well.

**Review by cijothomas — APPROVED** [review]
changelog please before merge.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs (hunk 1 above)
Comment (alanwest) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporter.cs:35]: Right, if #1504 is the pattern we want going forward, then I don't think we need to expose the classes. We could always re-expose the classes with a public constructor if we saw some need for that in the future.

**Pull request merged**